### PR TITLE
feat: create new methods with Config arguments for existing builders

### LINKED ISSUE #5835: Create `new` methods with Config arguments for existing builders
### Describe the feature

ref https://github.com/paradigmxyz/reth/pull/5834#discussion_r1432934073

We currently have `From` impls in the `reth-config` crate:
https://github.com/paradigmxyz/reth/blob/d673f5fbfbe3bf9e21d1b0d147d9facc39a580d2/crates/config/src/config.rs#L179-L181

Instead, we could create a method with this signature and remove the `From` impl:
```rust
    /// Creates a new [BodiesDownloaderBuilder] from the given [BodiesConfig].
    pub fn new(bodies_config: BodiesConfig) -> Self {
```

It would also make sense to come up with an easy to use method to replace `Config::network_config`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Part of #5835.

### CODE REVIEW

File: crates/net/downloaders/src/headers/reverse_headers.rs [hunk 1]
```diff
@@ -1087,6 +1088,19 @@ pub struct ReverseHeadersDownloaderBuilder {
     max_buffered_responses: usize,
 }
 
+impl ReverseHeadersDownloaderBuilder {
+    /// Creates a new `ReverseHeadersDownloaderBuilder` with configurations based on the provided
+    /// `HeadersConfig`.
```
Comment (Rjected) [inline crates/net/downloaders/src/headers/reverse_headers.rs:1093]:
```suggestion
    /// Creates a new [ReverseHeadersDownloaderBuilder] with configurations based on the provided
    /// [HeadersConfig].
```

File: crates/net/downloaders/src/bodies/bodies.rs [hunk 2]
```diff
@@ -492,6 +493,21 @@ pub struct BodiesDownloaderBuilder {
     pub concurrent_requests_range: RangeInclusive<usize>,
 }
 
+impl BodiesDownloaderBuilder {
+    /// Creates a new `BodiesDownloaderBuilder` with configurations based on the provided
+    /// `BodiesConfig`.
```
Comment (Rjected) [inline crates/net/downloaders/src/bodies/bodies.rs:498]:
```suggestion
    /// Creates a new [BodiesDownloaderBuilder] with configurations based on the provided
    /// [BodiesConfig].
```

**Review by Rjected — CHANGES_REQUESTED** [review]
just some doc nits, otherwise this looks good to me. Going to keep the issue open though, since we might still want something to replace `Config::network_config`.

**Comment by tcoratger** [comment]
> just some doc nits, otherwise this looks good to me. Going to keep the issue open though, since we might still want something to replace `Config::network_config`.

Fine just modified the description not to close the issue automatically.

**Review by Rjected — APPROVED** [review]
looks good to me, thanks!

**Review by mattsse — APPROVED** [review]
great!

**Pull request merged**